### PR TITLE
Consider a trailing dot when resolve DNS with search domains

### PR DESCRIPTION
**Pull request by ikhoon**
Motivation:

There was a report from LY internally where DNS resolver warned for `NXDomain` unexpectedly.
```java
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException: Empty label is not a legal name
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
        ...
	at com.linecorp.armeria.internal.client.dns.SearchDomainDnsResolver.resolve0(SearchDomainDnsResolver.java:99)
	at com.linecorp.armeria.internal.client.dns.SearchDomainDnsResolver.resolve(SearchDomainDnsResolver.java:88)
	at com.linecorp.armeria.internal.client.dns.HostsFileDnsResolver.resolve(HostsFileDnsResolver.java:130)
	at com.linecorp.armeria.internal.client.dns.DefaultDnsResolver.resolveOne(DefaultDnsResolver.java:89)
	at com.linecorp.armeria.internal.client.dns.DefaultDnsResolver.resolve(DefaultDnsResolver.java:81)
	at com.linecorp.armeria.client.endpoint.dns.DnsEndpointGroup.sendQueries(DnsEndpointGroup.java:155)
	at com.linecorp.armeria.client.endpoint.dns.DnsEndpointGroup.lambda$sendQueries$3(DnsEndpointGroup.java:173)
       ...
Caused by: java.lang.IllegalArgumentException: Empty label is not a legal name
  at java.base/java.net.IDN.toASCIIInternal(IDN.java:284)
  at java.base/java.net.IDN.toASCII(IDN.java:123)
  at java.base/java.net.IDN.toASCII(IDN.java:152)
  at com.linecorp.armeria.internal.client.dns.DnsQuestionWithoutTrailingDot.<init>(DnsQuestionWithoutTrailingDot.java:53)
  at com.linecorp.armeria.internal.client.dns.DnsQuestionWithoutTrailingDot.of(DnsQuestionWithoutTrailingDot.java:48)
  at com.linecorp.armeria.internal.client.dns.SearchDomainDnsResolver$SearchDomainQuestionContext.newQuestion(SearchDomainDnsResolver.java:190)
  at com.linecorp.armeria.internal.client.dns.SearchDomainDnsResolver$SearchDomainQuestionContext.nextQuestion0(SearchDomainDnsResolver.java:177)
  at com.linecorp.armeria.internal.client.dns.SearchDomainDnsResolver$SearchDomainQuestionContext.nextQuestion(SearchDomainDnsResolver.java:150)
  at com.linecorp.armeria.internal.client.dns.SearchDomainDnsResolver.lambda$resolve0$1(SearchDomainDnsResolver.java:103)
  ... 19 common frames omitted
```

The NX domain has a trailing dot and search domains start with a dot (`.`). 
As a result, `example.com..search.domain` was made and rejected by `java.net.IDN`

Modifications:

- Remove a leading dot from the normalized search domains.
- Infix a dot when a hostname does not have a trailing dot.

Result:

DNS resolver now correctly adds search domains for hostnames with trailing dots.

### CODE REVIEW

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Review by jrhee17 — APPROVED** [review]
👍 👍

File: core/src/main/java/com/linecorp/armeria/internal/client/dns/DefaultDnsResolver.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ private CompletableFuture<List<DnsRecord>> resolveOne(DnsQuestionContext ctx, Dn
         });
         future.handle((unused0, unused1) -> {
             // Maybe cancel the timeout scheduler.
-            ctx.cancel();
+            ctx.cancelScheduler();
```
Comment (minwoox) [inline core/src/main/java/com/linecorp/armeria/internal/client/dns/DefaultDnsResolver.java:99]: Can't we just call `ctx.setComplete();` to set the `complete` to true?

**Review by minwoox — APPROVED** [review]
Still looks good 👍

**Pull request merged**